core/src/test/java/org/apache/accumulo/core/security/crypto/CryptoTest.java

```diff
@@ -357,14 +363,209 @@ public void testKeyWrapAndUnwrap() throws NoSuchAlgorithmException, NoSuchPaddin
+    CryptoModuleParameters.incrementIV(testIv3, testIv3.length - 1);
+    Assert.assertArrayEquals(testIv3, new byte[] {(byte)0x00, (byte)0x00, (byte)0x01});
+
+  }
+
+  /**
+   * Used in AESGCM unit tests to encrypt data. Uses MARKER_STRING and MARKER_INT
+   *
+   * @param conf
+   *          The accumulo configuration
+   * @param initVector
+   *          The IV to be used in encryption
+   * @return the encrypted string
+   * @throws IOException
+   *           if DataOutputStream fails
+   */
+  private static byte[] testEncryption(AccumuloConfiguration conf, byte[] initVector) throws IOException {
+    CryptoModuleParameters params = CryptoModuleFactory.createParamsObjectFromAccumuloConfiguration(conf);
+    CryptoModule cryptoModule = CryptoModuleFactory.getCryptoModule(conf);
+    params.getAllOptions().put(Property.CRYPTO_WAL_CIPHER_SUITE.getKey(), "AES/GCM/NoPadding");
+    params.setInitializationVector(initVector);
+
+    /*
+     * Now lets encrypt this data!
+     */
+    ByteArrayOutputStream encryptedByteStream = new ByteArrayOutputStream();
+    params.setPlaintextOutputStream(new NoFlushOutputStream(encryptedByteStream));
+    params = cryptoModule.getEncryptingOutputStream(params);
+    DataOutputStream encryptedDataStream = new DataOutputStream(params.getEncryptedOutputStream());
+    encryptedDataStream.writeUTF(MARKER_STRING);
+    encryptedDataStream.writeInt(MARKER_INT);
+    encryptedDataStream.close();
+    byte[] encryptedBytes = encryptedByteStream.toByteArray();
+    return (encryptedBytes);
+  }
+
+  /**
+   * Used in AESGCM unit tests to decrypt data. Uses MARKER_STRING and MARKER_INT
+   *
+   * @param conf
+   *          The accumulo configuration
+   * @param encryptedBytes
+   *          The encrypted bytes
+   * @return 0 if data is incorrectly decrypted, 1 if decrypted data matches input, 2 if data integrity was compromised
+   * @throws IOException
+   *           if DataInputStream fails
+   * @throws AEADBadTagException
+   *           if the encrypted stream has been modified
+   */
+  private static Integer testDecryption(AccumuloConfiguration conf, byte[] encryptedBytes) throws IOException, AEADBadTagException {
+    CryptoModuleParameters params = CryptoModuleFactory.createParamsObjectFromAccumuloConfiguration(conf);
+    CryptoModule cryptoModule = CryptoModuleFactory.getCryptoModule(conf);
+    ByteArrayInputStream decryptedByteStream = new ByteArrayInputStream(encryptedBytes);
+    params.setEncryptedInputStream(decryptedByteStream);
+    params = cryptoModule.getDecryptingInputStream(params);
+    DataInputStream decryptedDataStream = new DataInputStream(params.getPlaintextInputStream());
+
+    String utf;
+    Integer in;
+    try {
+      utf = decryptedDataStream.readUTF();
+      in = decryptedDataStream.readInt();
+    } catch (IOException e) {
+      final String message = "javax.crypto.AEADBadTagException:";
+      if (e.getMessage().substring(0, message.length()).equals(message)) {
```

Comment:
Do you know if e.getCause() would return an exception of type AEADBadTagException?  If so that may be a cleaner way to check for this.